WordPressKit/WordPressComOAuthClient.swift

```diff
@@ -802,4 +844,25 @@ private extension WordPressComOAuthClient {
             .append(path: "/wp-login.php")
             .query(name: "action", value: action.rawValue)
     }
+
+    enum SocialSignInAction: String {
+        case sendOTPViaSMS = "send-sms-code-endpoint"
+        case authenticate = "social-login-endpoint"
+        case authenticateWith2FA = "two-step-authentication-endpoint"
+
+        var queryItems: [URLQueryItem] {
+            var items = [URLQueryItem(name: "action", value: rawValue)]
+            if self == .authenticate || self == .authenticateWith2FA {
+                items.append(URLQueryItem(name: "version", value: "1.0"))
+            }
+            return items
+        }
+    }
```

Comment:
What do you think of adding unit tests for this conditional logic?

```swift
XCTAssertEqual(sendOTPViaSMS.queryItems.count, 1)
XCTAssertTrue(sendOTPViaSMS.queryItems.contains[URLQueryItem(name: "action", value: "send-sms-code-endpoint")])
...
```

Admittedly, it is trivial logic, but one that could be hard to find if broken.

Reply:
[The existing tests](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/d182249d336ce5e3d34e830470e93edf82cf0ab7/WordPressKitTests/WordPressComOAuthClientTests.swift#L12-L13), which stub URL requests by their full URL, have covered these query parameters. Removing either one of these two query will cause the existing tests to fail. So, I think we are all good in terms of coverage? But let me know if you think we should add dedicated tests for these query parameters specifically.